tests/test_sourmash.py

```diff
@@ -0,0 +1,156 @@
+        )
+
+
+def test_bad_query_or_subject(
+    tmp_path: str, input_genomes_tiny: Path, sourmash_targets_compare_indir: Path
+) -> None:
+    """Mismatch between query or subject FASTA in sourmash compare output and commandline."""
+    tmp_db = Path(tmp_path) / "new.sqlite"
+    assert not tmp_db.is_file()
+
+    with pytest.raises(
+        SystemExit,
+        match=(
+            "ERROR: Given --query-fasta .*/MGV-GENOME-0266457.fna"
+            " but query in sourmash compare filename was MGV-GENOME-0264574"
+        ),
+    ):
+        private_cli.log_sourmash(
+            database=tmp_db,
+            query_fasta=input_genomes_tiny / "MGV-GENOME-0266457.fna",
+            subject_fasta=input_genomes_tiny / "MGV-GENOME-0266457.fna",
+            compare=sourmash_targets_compare_indir
+            / "MGV-GENOME-0264574_vs_MGV-GENOME-0266457.csv",
+        )
+
+    with pytest.raises(
+        SystemExit,
+        match=(
+            "ERROR: Given --subject-fasta .*/MGV-GENOME-0264574.fas"
+            " but subject in sourmash compare filename was MGV-GENOME-0266457"
+        ),
+    ):
+        private_cli.log_sourmash(
+            database=tmp_db,
+            query_fasta=input_genomes_tiny / "MGV-GENOME-0264574.fas",
+            subject_fasta=input_genomes_tiny / "MGV-GENOME-0264574.fas",
+            compare=sourmash_targets_compare_indir
+            / "MGV-GENOME-0264574_vs_MGV-GENOME-0266457.csv",
+        )
+
+
+def test_logging_sourmash(
+    tmp_path: str,
+    input_genomes_tiny: Path,
+    sourmash_targets_compare_indir: Path,
+    dir_sourmash_results: Path,
+) -> None:
+    """Check can log a sourmash comparison to DB."""
+    tmp_db = Path(tmp_path) / "new.sqlite"
+    assert not tmp_db.is_file()
+
+    tool = tools.get_sourmash()
+
+    private_cli.log_configuration(
+        database=tmp_db,
+        method="sourmash",
+        program=tool.exe_path.stem,
+        version=tool.version,
+        mode=method_sourmash.MODE,
+        kmersize=method_sourmash.KMER_SIZE,
+        extra=method_sourmash.EXTRA,
+        create_db=True,
+    )
+    private_cli.log_genome(
+        database=tmp_db,
+        fasta=[
+            input_genomes_tiny / "MGV-GENOME-0264574.fas",
+            input_genomes_tiny / "MGV-GENOME-0266457.fna",
+        ],
+    )
+
+    private_cli.log_sourmash(
+        database=tmp_db,
+        query_fasta=input_genomes_tiny / "MGV-GENOME-0264574.fas",
+        subject_fasta=input_genomes_tiny / "MGV-GENOME-0266457.fna",
+        compare=sourmash_targets_compare_indir
+        / "MGV-GENOME-0264574_vs_MGV-GENOME-0266457.csv",
+    )
+
+    # Check the recorded comparison values
+    session = db_orm.connect_to_db(tmp_db)
+    assert session.query(db_orm.Comparison).count() == 1
+    comp = session.query(db_orm.Comparison).one()
+    query = "689d3fd6881db36b5e08329cf23cecdd"  # MGV-GENOME-0264574.fas
+    subject = "78975d5144a1cd12e98898d573cf6536"  # MGV-GENOME-0266457.fna
```

Comment:
You can confirm ``query == comp.query_hash`` here (and for subject), or remove those two lines and use ``comp.query_hash`` below?